cirq-core/cirq/ops/boolean_hamiltonian_test.py

```diff
@@ -87,6 +87,69 @@ def test_circuit(boolean_str):
     np.testing.assert_array_equal(actual, expected)
 
 
+@pytest.mark.parametrize(
+    'boolean_str',
+    [
+        'x0',
+        '~x0',
+        'x0 ^ x1',
+        'x0 & x1',
+        'x0 | x1',
+        'x0 & x1 & x2',
+        'x0 & x1 & ~x2',
+        'x0 & ~x1 & x2',
+        'x0 & ~x1 & ~x2',
+        '~x0 & x1 & x2',
+        '~x0 & x1 & ~x2',
+        '~x0 & ~x1 & x2',
+        '~x0 & ~x1 & ~x2',
+        'x0 ^ x1 ^ x2',
+        'x0 | (x1 & x2)',
+        'x0 & (x1 | x2)',
+        '(x0 ^ x1 ^ x2) | (x2 ^ x3 ^ x4)',
+        '(x0 ^ x2 ^ x4) | (x1 ^ x2 ^ x3)',
+        'x0 & x1 & (x2 | x3)',
+        'x0 & ~x2',
+        '~x0 & x2',
+        'x2 & ~x0',
+        '~x2 & x0',
+        '(x2 | x1) ^ x0',
+    ],
+)
+def test_gate_circuit(boolean_str):
+    boolean_expr = sympy_parser.parse_expr(boolean_str)
```

Comment:
This is quite a bit of repeated code here. Would it be possible to remove some duplication?